packages/serverpod_test/lib/src/with_serverpod.dart

```diff
@@ -49,6 +63,50 @@ typedef TestClosure<T> = void Function(
   TestSession testSession,
 );
 
+/// Internal helper extension for the [DatabaseTestConfig] class.
+extension ConfigRuntimeChecks on DatabaseTestConfig {
+  /// Returns true if rollbacks are disabled.
+  bool get areRollbacksDisabled => _rollbackDatabase == _RollbackDatabase.never;
+}
+
+/// Configuration object for how `withServerpod` should handle the database.
+class DatabaseTestConfig {
+  final _RollbackDatabase _rollbackDatabase;
+
+  /// Creates a new database test configuration.
+  DatabaseTestConfig._({
+    _RollbackDatabase rollbackDatabase = _RollbackDatabase.afterEach,
+  }) : _rollbackDatabase = rollbackDatabase;
+
+  /// Rolls back the database after each test. This is the default.
+  /// Will throw an [InvalidConfigurationException] if several calls to `transaction` are made concurrently
+  /// since this is not supported by the test tools.
+  /// In this case, disable rolling back the database by setting `databaseTestConfig` to `DatabaseTestConfig.rollbacksDisabled()`.
+  factory DatabaseTestConfig.rollbackAfterEach() {
+    return DatabaseTestConfig._(
+      rollbackDatabase: _RollbackDatabase.afterEach,
+    );
+  }
```

Comment:
Discussion: This works in the case when we configure a single thing. But since we have created a TestConfig object we will probably want to add more things to it later?

How do you see this interface scale in that case?

I'm a bit concerned that the factory methods will be obselete as soon as we add another option.

Reply:
Could be, yea. I'm not sure about this either. I will just revert back to the enum to reduce the scope of this PR.

Reply:
100408c